testing/main_integration_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// +build integration
+
+package testing
```

Comment:
Didn't we agree not to put Go code in this repository?

Reply:
I think putting this in a different repo already know complicates things unnecessary, but I'm with you that it doesn't have to be here. The way I imagine it in the future is that assuming we have an API to publish new packages, this "testing" for the specific package would be run during the push or a building phase.

Reply:
I see. I think it would be valuable to take some time and write down a short paragraph on the service look in the future.

Reply:
The code we have here, allows to take any directory with a list of packages, setup the stack and test if the packages can be installed into the stack. The package-storage itself needs it make sure no "invalid" packages are added to the package-storage. It is basically part of the validation.

Now the question, where this code should go in the future? Is that something that `elastic-package` should support and we just used it here?

My suggestion is that we get it into package-storage for now until we figured out where else to place it.

About the no code policy: I think the important part around the no code is that the code does NOT modify the packages itself in any way.

Reply:
> My suggestion is that we get it into package-storage for now until we figured out where else to place it.

Agree, but at some point in time this model won't scale anymore and probably we'll have to switch to artifactory or sth else. Simply, the repository will be too big to checkout. Maybe we should start considering the possible migration to a different storage. The less code is in this repository, the simpler migration will be.